test/e2e/network/endpointslice.go

```diff
@@ -161,6 +152,62 @@ var _ = SIGDescribe("EndpointSlice", func() {
 		}
 	})
 
+	ginkgo.It("should maintain EndpointSlices for a Service after a quick delete and recreation", func() {
+		svcConfig := &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "example-empty-selector",
+			},
+			Spec: v1.ServiceSpec{
+				Selector: map[string]string{
+					"does-not-match-anything": "endpoints-and-endpoint-slices-should-still-be-created",
+				},
+				Ports: []v1.ServicePort{{
+					Name:     "example",
+					Port:     80,
+					Protocol: v1.ProtocolTCP,
+				}},
+			},
+		}
+		svc := createServiceReportErr(cs, f.Namespace.Name, svcConfig)
+
+		// Expect EndpointSlice resource to be created.
+		var endpointSlice discoveryv1beta1.EndpointSlice
+		if err := wait.PollImmediate(2*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+			endpointSliceList, err := cs.DiscoveryV1beta1().EndpointSlices(svc.Namespace).List(context.TODO(), metav1.ListOptions{
+				LabelSelector: "kubernetes.io/service-name=" + svc.Name,
+			})
+			if err != nil {
+				return false, err
+			}
+			if len(endpointSliceList.Items) == 0 {
+				return false, nil
+			}
+			endpointSlice = endpointSliceList.Items[0]
+			return true, nil
+		}); err != nil {
+			framework.Failf("No EndpointSlice found for Service %s/%s: %s", svc.Namespace, svc.Name, err)
+		}
+
+		ensureValidEmptyEndpointSlice(endpointSlice)
+
+		err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err, "error deleting Service")
+
+		svc = createServiceReportErr(cs, f.Namespace.Name, svcConfig)
+
+		// Expect EndpointSlice resource to exist after 5 second delay.
+		time.Sleep(5 * time.Second)
```

Comment:
this is a bit scary, the good thing is that it will only have false positives, I means the slice is created later than that

Reply:
That's a good point, do you think there's a safer/more reliable way to cover this with an e2e test? I think this is already covered fairly well with unit tests so I could get rid of this entirely.

Reply:
Yeah, or integration or unit, but this e2e does not seems worthy. Imho this is good without the e2e

Reply:
Good call, updated to remove the e2e test entirely.